src/collect_columns/collect_columns.py

```diff
@@ -46,24 +47,20 @@ def collect_columns(count_tables: List[Path], feature_column: int,
     # If no sample names are given use table basenames instead.
     if names is None:
         names = [path.name for path in count_tables]
-    # Convert tables_have_headers to conform with the pandas parameter.
```

Comment:
If names is not None, it would be nice to add a check to see if `len(count_tables) == len(names)`

Reply:
Do you suggest throwing an error if this check is not matched or falling back on the file names like when it's None?

Reply:
Throwing an error is more user friendly than silently correcting a user input error. The last one will lead to unexpected behaviour (from the user's perspective)